### ISSUE TITLE
Cannot use the function to detect Objects References

### ISSUE THREAD

**Issue by hasenclever** [DaxStudio/DaxStudio#388]
**Describe the bug**
A open the DAX Studio, connect to a PBIX file and trying to find out any objects references gives an error that tells me that: "There is a dependence node not allowed".

**To Reproduce**
1. Open Dax Studio
2. Connect to any PBIX
3. Click on any object with the right mouse button
4. Select SHOW OBJECTS THAT REFERENCE (Table or Measure or Column)

**Expected behavior**
List of objects that reference the chosen one.

**Screenshots**
![image](https://user-images.githubusercontent.com/31460121/88309604-d715fc80-cd0e-11ea-8a6c-9751668e3fed.png)

**Comment by dgosbell** [comment]
> "There is a dependence node not allowed"

This is not an error message generated by DAX Studio, so this must be a Power BI Desktop error. Does this happen with multiple files or just this one?

**Comment by hasenclever** [comment]
E

> > "There is a dependence node not allowed"
> 
> This is not an error message generated by DAX Studio, so this must be a Power BI Desktop error. Does this happen with multiple files or just this one?

Just test it and there some files where it does not happen. Nevertheless, the error appeared on a new PBIX I was creating. And it also appears on a old file where it use to work.

If you ask me, it is a bug from the new integration with the EXTERNAL TOOLS option on PBI Desktop.

For instance, on my computer, my software is installed on drive D: and I could not have DAX Studio installed on D, even though I changed the parameter of the path on the Power Common file. To able to new EXTERNAL TOOL button to work and open DAX Studio I had to install it on C:.

And I just made a test. On a PBIX file that was created last week it presents me the error. I created a new one today, copied all the objects from the oldest file to the new one and works.

Just for comparison sake, I don't have this behavior on the Tabular Editor.

**Comment by dgosbell** [comment]
> If you ask me, it is a bug from the new integration with the EXTERNAL TOOLS option on PBI Desktop.

I do not think the external tools feature itself would not be causing this. But it could be something to do with the enhanced metadata format which is what is enabling editing tools like Tabular Editor and ALM Toolkit.

> For instance, on my computer, my software is installed on drive D: and I could not have DAX Studio installed on D, even though I changed the parameter of the path on the Power Common file. To able to new EXTERNAL TOOL button to work and open DAX Studio I had to install it on C:.

That's just a bug in the installer that is not updating for custom install paths. That will be fixed in the next release. Changing the pbitool.json file should work. I've just tested the new installer on my machine when I install to D: and it works fine. I should be able to get this new release published in the next day or so.

> And I just made a test. On a PBIX file that was created last week it presents me the error. I created a new one today, copied all the objects from the oldest file to the new one and works.

Have you tried running the ALM toolkit against these two files to see if they are completely identical? 

It's hard for me to say what is going on here without a way of reproducing the error.

> Just for comparison sake, I don't have this behavior on the Tabular Editor.

This error is probably specific to the DISCOVER_CALC_DEPENDENCIES DMV. I don't believe Tabular Editor uses the DISCOVER_CALC_DEPENDENCIES DMV, I think it builds its own dependency graph.

**Comment by hasenclever** [comment]
You are right it has to do with the metada function. The problem is, once you activated the preview feature of the meta data, the problem occurs on the DAX Studio. And to solve it, it is not enough just to deactivate the PBI feature. You have to create the file all over again,

Thanks.

**Comment by hasenclever** [comment]
By the way, let me tell, that my effort in trying to solve is because I just cannot stand to cannot use the DAX Studio anymore. Excellent job guys.

**Comment by dgosbell** [comment]
Yes, if I create a new file I get this issue too. I'll ask the Power BI team if they know about this issue.

**Comment by hasenclever** [comment]
Thanks manOn Jul 24, 2020 5:00 PM, Darren Gosbell <notifications@github.com> wrote:
Yes, if I create a new file I get this issue too. I'll ask the Power BI team if they know about this issue.

**Comment by hasenclever** [comment]
@dgosbell can you give some update please?

**Comment by dgosbell** [comment]
It's a Microsoft bug, there is nothing I can really do about it. I've let them know about it and they said they would look into it. In the mean time you could try using the dependency view in Tabular Editor as I think it might build it's own dependency graph.

**Comment by chrolivier** [comment]
@dgosbell I'm experiencing the same issue when using the $SYSTEM.DISCOVER_CALC_DEPENDENCY statement in a project of mine. Where did you raise the issue with Microsoft? Maybe I can upvote or something to help add some urgency to the issue?

**Comment by dgosbell** [comment]
@chrolivier - Yes, I would expect that everyone trying to query DISCOVER_CALC_DEPENDENCY when connect to a model using the enahanced metadata would hit this issue.

> Where did you raise the issue with Microsoft? Maybe I can upvote or something to help add some urgency to the issue?

I emailed some of the PMs on the engine team directly. I'm not aware of any public issue for this that you could vote on.

**Comment by chrolivier** [comment]
I'm a registered MS partner (for what it's worth) and I've built a tool that relies heavily on this functionality (https://www.sinapse.io/pbi-spy). Without it I'm hosed, so let me know if you need another voice to raise the profile of the issue.

**Comment by dgosbell** [comment]
@chrolivier - so raising a support ticket with Microsoft would be the one way of increasing the visibility with the dev team

**Comment by danning1234** [comment]
Does anyone have updates on this issue?

**Comment by dgosbell** [comment]
I did ask the product team if there were any updates on this a few days ago. It's still on their list, but no one has had a look at it yet.
